### PR TITLE
LegalHold APIs (Set, Get) with Args, Response.

### PR DESCRIPTION
**Pull request by BigUstad**
Functional Test, Examples

### CODE REVIEW

**Review by kannappanr — APPROVED** [review]
LGTM

File: Minio/ApiEndpoints/IObjectOperations.cs [hunk 1]
```diff
@@ -27,6 +27,26 @@ namespace Minio
...
     public interface IObjectOperations
     {
+        /// <summary>
+        /// Get the configuration object for Legal Hold Status 
+        /// </summary>
+        /// <param name="args">GetObjectLegalHoldArgs Arguments Object which has object identifier information - bucket name, object name, version ID</param>
+        /// <param name="cancellationToken">Optional cancellation token to cancel the operation </param>
+        /// <returns> True if Legal Hold is ON, false otherwise  </returns>
+        /// <exception cref="InvalidBucketNameException">When bucketName is invalid</exception>
+        /// <exception cref="InvalidObjectNameException">When bucketName is invalid</exception>
```
Comment (poornas) [inline Minio/ApiEndpoints/IObjectOperations.cs:37]:
```suggestion
        /// <exception cref="InvalidObjectNameException">When objectName is invalid</exception>
```

File: Minio/ApiEndpoints/IObjectOperations.cs [hunk 2]
```diff
@@ -27,6 +27,26 @@ namespace Minio
...
+        Task<bool> GetObjectLegalHoldAsync(GetObjectLegalHoldArgs args, CancellationToken cancellationToken = default(CancellationToken));
+
+        /// <summary>
+        /// Get the configuration object for Legal Hold Status 
+        /// </summary>
+        /// <param name="args">SetObjectLegalHoldArgs Arguments Object which has object identifier information - bucket name, object name, version ID and the status (ON/OFF) of legal-hold</param>
+        /// <param name="cancellationToken">Optional cancellation token to cancel the operation </param>
+        /// <returns> True if Legal Hold is ON, false otherwise  </returns>
+        /// <exception cref="InvalidBucketNameException">When bucketName is invalid</exception>
+        /// <exception cref="InvalidObjectNameException">When bucketName is invalid</exception>
```
Comment (poornas) [inline Minio/ApiEndpoints/IObjectOperations.cs:47]:
```suggestion
        /// <exception cref="InvalidObjectNameException">When objectName is invalid</exception>
```

File: Minio/ApiEndpoints/ObjectOperations.cs [hunk 3]
```diff
@@ -177,6 +177,39 @@ public async Task<string> PresignedPutObjectAsync(PresignedPutObjectArgs args)
...
         }
 
+        /// <summary>
+        /// Get the configuration object for Legal Hold Status 
+        /// </summary>
+        /// <param name="args">GetObjectLegalHoldArgs Arguments Object which has object identifier information - bucket name, object name, version ID</param>
+        /// <param name="cancellationToken">Optional cancellation token to cancel the operation </param>
+        /// <returns> True if Legal Hold is ON, false otherwise  </returns>
+        /// <exception cref="InvalidBucketNameException">When bucketName is invalid</exception>
+        /// <exception cref="InvalidObjectNameException">When bucketName is invalid</exception>
```
Comment (poornas) [inline Minio/ApiEndpoints/ObjectOperations.cs:187]:
```suggestion
        /// <exception cref="InvalidObjectNameException">When objectName is invalid</exception>
```

**Review by poornas — CHANGES_REQUESTED** [review]

File: Minio/ApiEndpoints/IObjectOperations.cs (hunk 1 above)
Comment (BigUstad) [reply Minio/ApiEndpoints/IObjectOperations.cs]: It is corrected. Thx. PTAL.

File: Minio/ApiEndpoints/ObjectOperations.cs (hunk 3 above)
Comment (BigUstad) [reply Minio/ApiEndpoints/ObjectOperations.cs]: Changed. PTAL.

File: Minio/ApiEndpoints/IObjectOperations.cs (hunk 2 above)
Comment (BigUstad) [reply Minio/ApiEndpoints/IObjectOperations.cs]: Changed. PTAL.

**Comment by kannappanr** [comment]
@poornas PTAL

File: Minio/ApiEndpoints/IObjectOperations.cs [hunk 4]
```diff
@@ -27,6 +27,26 @@ namespace Minio
...
+        /// </summary>
+        /// <param name="args">GetObjectLegalHoldArgs Arguments Object which has object identifier information - bucket name, object name, version ID</param>
+        /// <param name="cancellationToken">Optional cancellation token to cancel the operation </param>
+        /// <returns> True if Legal Hold is ON, false otherwise  </returns>
+        /// <exception cref="InvalidBucketNameException">When bucketName is invalid</exception>
+        /// <exception cref="InvalidObjectNameException">When objectName is invalid</exception>
+        Task<bool> GetObjectLegalHoldAsync(GetObjectLegalHoldArgs args, CancellationToken cancellationToken = default(CancellationToken));
+
+        /// <summary>
+        /// Get the configuration object for Legal Hold Status
```
Comment (poornas) [inline Minio/ApiEndpoints/IObjectOperations.cs:41]:
```suggestion
        /// Set the configuration object for Legal Hold Status

**Review by poornas — APPROVED** [review]
LGTM

File: Minio/ApiEndpoints/IObjectOperations.cs (hunk 4 above)
Comment (BigUstad) [reply Minio/ApiEndpoints/IObjectOperations.cs]: Changed it. Tx.

File: Minio/Exceptions/MissingObjectLockConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
+/*
+ * MinIO .NET Library for Amazon S3 Compatible Cloud Storage, (C) 2017 MinIO, Inc.
```
Comment (kannappanr) [inline Minio/Exceptions/MissingObjectLockConfiguration.cs:2]:
```suggestion
 * MinIO .NET Library for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
```

**Review by kannappanr — APPROVED** [review]

**Pull request merged**